### ISSUE TITLE
Looping play head movements as really intended

### ISSUE THREAD

**Issue by Paul-Licameli** [audacity/audacity#1994]
The real intent for movements of the play head, with looping playback:

* Play starts at the left edge of selection, even if that is outside of the looping region.
 * If that is right of the region, or the loop region is disabled, then play just continues to the end and stops.
 * If that is left of the region or in it, play continues until the right edge of loop region, then jumps to the left.

In other words, the looping region is a "trap" for the play head.  Also:

* These rules are applied even if the play region is changed during play.
* The rules are applied even if the play region is toggled during play.

See also issue #1984.
